src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/Vector128.cs

```diff
@@ -171,6 +173,83 @@ public static Vector128<ulong> AsUInt64<T>(this Vector128<T> vector)
             return vector.As<T, ulong>();
         }
 
+        /// <summary>Reinterprets a <see cref="Vector2" /> as a new <see cref="Vector128{Single}" />.</summary>
+        /// <param name="value">The vector to reinterpret.</param>
+        /// <returns><paramref name="value" /> reinterpreted as a new <see cref="Vector128{Single}" />.</returns>
+        public static Vector128<float> AsVector128(this Vector2 value)
+        {
+            return new Vector4(value, 0.0f, 0.0f).AsVector128();
+        }
+
+        /// <summary>Reinterprets a <see cref="Vector3" /> as a new <see cref="Vector128{Single}" />.</summary>
+        /// <param name="value">The vector to reinterpret.</param>
+        /// <returns><paramref name="value" /> reinterpreted as a new <see cref="Vector128{Single}" />.</returns>
+        public static Vector128<float> AsVector128(this Vector3 value)
+        {
+            return new Vector4(value, 0.0f).AsVector128();
+        }
+
+        /// <summary>Reinterprets a <see cref="Vector4" /> as a new <see cref="Vector128{Single}" />.</summary>
+        /// <param name="value">The vector to reinterpret.</param>
+        /// <returns><paramref name="value" /> reinterpreted as a new <see cref="Vector128{Single}" />.</returns>
+        public static Vector128<float> AsVector128(this Vector4 value)
+        {
+            return Unsafe.As<Vector4, Vector128<float>>(ref value);
+        }
+
+        /// <summary>Reinterprets a <see cref="Vector{T}" /> as a new <see cref="Vector128{T}" />.</summary>
+        /// <typeparam name="T">The type of the vectors.</typeparam>
+        /// <param name="value">The vector to reinterpret.</param>
+        /// <returns><paramref name="value" /> reinterpreted as a new <see cref="Vector128{T}" />.</returns>
+        /// <exception cref="NotSupportedException">The type of <paramref name="value" /> (<typeparamref name="T" />) is not supported.</exception>
+        public static Vector128<T> AsVector128<T>(this Vector<T> value)
+            where T : struct
+        {
+            Debug.Assert(Vector<T>.Count >= Vector128<T>.Count);
```

Comment:
This currently assumes that `Vector<T>` won't ever be 64-bytes. This is `true` today, but the logic would need to change if we ever supported something less (like on `ARM`/`ARM64` where `Vector64<T>` is the minimum).